agent/exec/container/executor.go

```diff
@@ -41,12 +41,35 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 		}
 	}
 
+	// add v1 plugins to 'plugins'
 	addPlugins("Volume", info.Plugins.Volume)
 	// Add builtin driver "overlay" (the only builtin multi-host driver) to
 	// the plugin list by default.
 	addPlugins("Network", append([]string{"overlay"}, info.Plugins.Network...))
 	addPlugins("Authorization", info.Plugins.Authorization)
 
+	// retrieve v2 plugins
+	v2plugins, err := e.client.PluginList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// add v2 plugins to 'plugins'
+	for _, plgn := range v2plugins {
+		for _, typ := range plgn.Config.Interface.Types {
+			plgnTyp := typ.Capability
+			if typ.Capability == "volumedriver" {
+				plgnTyp = "Volume"
```

Comment:
On that note, the issue that we don't have a way of storing arbitrary plugins in the TaskSpec, from which to match them against the node on which `PluginFilter` is applied. For volumes and networks, this happens via a separate code path that allows for attaching to networks or mounting volumes. This is something we might want to build to support general plugins, and will require proto changes. This PR will work for volume and network v2 plugins.

cc @stevvooe